SystemServices/SystemServices.cpp

```diff
@@ -2227,40 +2227,58 @@ namespace WPEFramework {
         uint32_t SystemServices::setTimeZoneDST(const JsonObject& parameters,
                 JsonObject& response)
 	{
+		LOGERR("Entered to setTimeZoneDST");
 		bool resp = false;
 		if (parameters.HasLabel("timeZone")) {
 			std::string dir = dirnameOf(TZ_FILE);
 			std::string timeZone = "";
 			try {
 				timeZone = parameters["timeZone"].String();
+				int pos = timeZone.find("/");
 				if (timeZone.empty() || (timeZone == "null")) {
 					LOGERR("Empty timeZone received.");
-				} else {
-					if (!dirExists(dir)) {
-						std::string command = "mkdir -p " + dir + " \0";
-						Utils::cRunScript(command.c_str());
-					} else {
-						//Do nothing//
-					}
-					std::string oldTimeZoneDST = getTimeZoneDSTHelper();
-
-					FILE *f = fopen(TZ_FILE, "w");
-					if (f) {
-						if (timeZone.size() != fwrite(timeZone.c_str(), 1, timeZone.size(), f))
-							LOGERR("Failed to write %s", TZ_FILE);
-
-						fflush(f);
-						fsync(fileno(f));
-						fclose(f);
-						if (SystemServices::_instance)
-							SystemServices::_instance->onTimeZoneDSTChanged(oldTimeZoneDST,timeZone);
-						resp = true;
-					} else {
-						LOGERR("Unable to open %s file.\n", TZ_FILE);
-						populateResponseWithError(SysSrv_FileAccessFailed, response);
-						resp = false;
-					}
 				}
+				else if( (pos == string::npos) ||  ( (pos != string::npos) &&  (pos+1 == timeZone.length())  )   )
+				{
+					LOGERR("Invalid timeZone  Only Country received : %s . \n", timeZone.c_str());
+				}
+				else {
+					std::string path =ZONEINFO_DIR;
+					path += "/";
+					std::string country = timeZone.substr(0,pos);
+					std::string city = path+timeZone;
+					if( dirExists(path+country)  && Utils::fileExists(city.c_str()) ) 
+					{
+						if (!dirExists(dir)) {
+							std::string command = "mkdir -p " + dir + " \0";
+							Utils::cRunScript(command.c_str());
+						} else {
+							//Do nothing//
+						}
+						std::string oldTimeZoneDST = getTimeZoneDSTHelper();
+
+						FILE *f = fopen(TZ_FILE, "w");
+						if (f) {
+							if (timeZone.size() != fwrite(timeZone.c_str(), 1, timeZone.size(), f))
+								LOGERR("Failed to write %s", TZ_FILE);
+
+							fflush(f);
+							fsync(fileno(f));
+							fclose(f);
+							if (SystemServices::_instance)
+								SystemServices::_instance->onTimeZoneDSTChanged(oldTimeZoneDST,timeZone);
+							resp = true;
+						} else {
+							LOGERR("Unable to open %s file.\n", TZ_FILE);
+							populateResponseWithError(SysSrv_FileAccessFailed, response);
+							resp = false;
+						}
+						else{
+							LOGERR("Invalid timeZone  %s . \n", timeZone.c_str());
```

Comment:
Make this as generic message saying "invalid timezone format received. timezone should be in Olson format ("America/New_York")"

Reply:
Hi @anand-ky In this place the timezone format will be receive correctly(Ex: America/New_Yo ) but the timezone file will be miss . May I make the log as  "Invalid timezone received . The Timezone not avaibale in zoneinfo folder "